### ISSUE TITLE
How to define jsorm for situations like voiding an invoice

### ISSUE THREAD

**Issue by kharysharpe** [jsonapi-suite/jsorm#121]
Hi All,

Evaluating JSORM for a project. Read through the guide and not seeing any obvious JSORM way to to this. 

Background: User wants to cancel (void) a paid invoice, which effectively changes the status of the invoice, change associated payment(s) status, creates related journal entries (debits\credits) and then gives their customer a credit that they can use for future purchases.

If the api end point is /invoice/123/void

How would you go about doing this using JSORM?

**Comment by richmolj** [comment]
Hi @kharysharpe - JSORM supports JSONAPI, which is based on RESTful actions. I would think about how to refactor this into a RESTFUL action.

Here is a good talk on how to do so, and why it's a good idea:

https://youtu.be/HctYHe-YjnE

TLDR consider you are missing a domain object, for instance you vould POST  Cancellation object.

**Comment by wadetandy** [comment]
Hey @kharysharpe, thanks for opening an issue. I would tend to restructure my resource endpoints to be more “purely” RESTful. I would probably create something like /invoice_cancelations which accepts an invoice ID as an attribute. instead of a non-standard route on another resource, you’re CREATEing a new object type entirely (which just so happens to modify an invoice under the hood).

**Comment by wadetandy** [comment]
Looks like @richmolj and I are both sitting around near our email this Thanksgiving morning. At least we didn’t give you conflicting answers! 🤣

**Comment by kharysharpe** [comment]
Thanks for the quick replies.

@richmolj Thanks for the video, will check it out.

@wadetandy that's kinda what I was thinking, but (being new to RESTful mostly RPC) didn't know if that would have been a good approach.
